Fixed casting

Note: ftok defines -1 as error, but program handles mutex as if 0 is error.
(i.e. if (mutex >0) ... )

Index: micro-evtd-3.3.0/micro_evtd.c
===================================================================
--- micro-evtd-3.3.0.orig/micro_evtd.c	2008-06-22 23:43:24.000000000 +0200
+++ micro-evtd-3.3.0/micro_evtd.c	2008-06-22 23:49:50.000000000 +0200
@@ -1034,7 +1034,7 @@
 	int iRead;
 	int errno;
 	struct stat filestatus;
-	int* file;
+	int file;
 	int i;
 
 	errno = stat(micro_default, &filestatus);
@@ -1044,13 +1044,13 @@
 
 		// Has this file changed?
 		if (filestatus.st_mtime != tt_LastMicroAcess || iForce) {
-			file = (int*)open(micro_default, O_RDONLY);
+			file = open(micro_default, O_RDONLY);
 
-			if (file) {
-				iRead = read((int)file, &buff, 4095);
+			if (file >= 0) {
+				iRead = read(file, &buff, 4095);
 
 				// Dump the file pointer for others
-				close((int)file);
+				close(file);
 
 				if (iRead>0) {
 					/* Remove Windows End-of-Line formatting */
@@ -1101,7 +1101,7 @@
 	// Generate unique key.  This will fail with stock so let it be.
 	if ((mutex = ftok(micro_default, 'M')) == (key_t) -1) {
 	    printf("IPCS: Error, falling back to 'flock'\n");
-		mutex = (key_t)NULL;
+		mutex = 0;
 	}
 
 	// Parse any options
